File: C/ex01.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    float num1, num2, soma, sub, div, multi;
    char input[100];
    char *endptr;

    printf("Digite o 1o numero: ");
    fgets(input, sizeof(input), stdin);
    num1 = strtof(input, &endptr);

    if (input[0] == '\n' || *endptr != '\n') {
        printf("Entrada invalida para o primeiro numero. Por favor, digite um numero valido.\n");
        return 1;
    }

    printf("Digite o 2o numero: ");
    fgets(input, sizeof(input), stdin);
    num2 = strtof(input, &endptr);

    if (input[0] == '\n' || *endptr != '\n') {
        printf("Entrada invalida para o segundo numero. Por favor, digite um numero valido.\n");
        return 1;
    }

    if (num2 == 0) {
        printf("Impossivel dividir por zero.\n");
        return 1;
    }

    soma = num1 + num2;
    sub = num1 - num2;
    div = num1 / num2;
    multi = num1 * num2;

    printf("Soma: %.2f\n", soma);
    printf("Subtracao: %.2f\n", sub);
    printf("Divisao: %.2f\n", div);
    printf("Multiplicacao: %.2f\n", multi);

    return 0;
}

File: Java/ex01.java
import java.util.Scanner;

public class ex01 {
    public static void main(String[] args){    
        double soma, sub, div, multi;
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o primeiro numero: ");
        double n1 = scanner.nextDouble();
        System.out.print("Digite o segundo numero: ");
        double n2 = scanner.nextDouble();
    
        soma = n1 + n2;
        sub = n1 - n2;
        div = n1 / n2;
        multi = n1 * n2;
        
        System.out.println("Soma: "+ soma);
        System.out.println("Subtracao: "+ sub);
        System.out.println("Divisao: "+ div);
        System.out.println("Multiplicacao: "+ multi);
        
        scanner.close();
    }
}
